internal/client/client.go

```diff
@@ -127,6 +181,14 @@ func checkRetryPolicy(ctx context.Context, resp *http.Response, err error) (bool
 		return false, nil
 	}
 
+	// If the request is forbidden, no need to retry the request. Print
+	// out the error and stop retrying.
+	if resp.StatusCode == http.StatusForbidden {
+		body, _ := io.ReadAll(resp.Body)
+
+		return false, fmt.Errorf("status_code=%d, error=%w, body=%s", resp.StatusCode, err, body)
+	}
```

Comment:
I added this because when testing a failure case, the provider tried for almost 2 minutes to make the request before failing. 403s may as well fail early